splash/browser_tab.py

```diff
@@ -674,32 +675,52 @@ def html(self):
         self.store_har_timing("_onHtmlRendered")
         return result
 
-    def png(self, width=None, height=None, b64=False, render_all=False,
-            scale_method=None):
-        """ Return screenshot in PNG format """
-        self.logger.log("Getting PNG: width=%s, height=%s,"
-                        " render_all=%s, scale_method=%s" %
-                        (width, height, render_all, scale_method), min_level=2)
+    @contextmanager
```

Comment:
Yes, but the try/finally contains only a rollback of setViewportSize, so the only difference between contextmanager/yield/with and function/return is when that happens. It seems that it is ok to do it either way and contextmanager can be dropped, or can it not?